lib/registration.pm

```diff
@@ -624,7 +624,12 @@ sub install_docker_when_needed {
         die 'Docker is not pre-installed.' if zypper_call('se -x --provides -i docker | grep docker', allow_exit_codes => [0, 1]);
     }
     else {
-        add_suseconnect_product('sle-module-containers') if is_sle('15+');
+        if (is_sle('<15')) {
+            assert_script_run('zypper se docker || zypper -n ar -f http://download.suse.de/ibs/SUSE:/SLE-12:/Update/standard/SUSE:SLE-12:Update.repo');
```

Comment:
zypper ref , also don't you need to accept key from new repo ?

Reply:
apparently not -> http://lord.arch/tests/755#step/docker/10

Reply:
Please don't it in this way. QAM tests have already container module added during installation.
See https://openqa.suse.de/tests/1604714  MRU_ADDONS=sdk,tcm,wsm,we,contm
So basically there is no need to add special repo for SLE12-SP3(not sure about SP4).

Reply:
@czerw, please be aware that the repo will only be added if package docker is not found. When container module is already added, the repo will not be added. So it may work as expected.

But, I am still skeptical of this change. Now the reason why we need to add that repo is shadowed (to trick that we are using an image with unregistered SUT. An scenario where the customer also could not install docker). This tricky added repo may affect the following test modules on this test suite and people may forget about this and will probably understand/change this wrong. I would still prefer to use an image with registered SUT, so we avoid false negatives and confusion.

Reply:
@czerw @SergioAtSUSE I share your concerns but after your PR was already enabling docker on SLE12SP4 I will merge my PR in the hope to fix the test for now for SLE12SP4 as well … or actually to prevent failure. But we can keep https://progress.opensuse.org/issues/33184 open for now and discuss there.